pkg/apis/core/validation/validation.go

```diff
@@ -3551,11 +3551,28 @@ func ValidatePreferredSchedulingTerms(terms []core.PreferredSchedulingTerm, fldP
 	return allErrs
 }
 
+func validatePodAffinityTermNamespaceSelector(podAffinityTerm core.PodAffinityTerm, fldPath *field.Path) field.ErrorList {
+	nsSelector := podAffinityTerm.NamespaceSelector
+	if nsSelector == nil {
+		return nil
+	}
+
+	allErrs := unversionedvalidation.ValidateLabelSelector(nsSelector, fldPath)
+	for i, m := range nsSelector.MatchExpressions {
```

Comment:
Let's validate the values too

Reply:
Values have been validated in L3560 (ValidateLabelSelector -> ValidateLabelSelectorRequirement), although a bit dulicately.